src/util/ParseableDuration.h

```diff
@@ -39,26 +65,24 @@ class ParseableDuration {
       };
 
       if (unit == "ns") {
-        result = toDuration.template operator()<nanoseconds>(amount);
+        return toDuration.template operator()<nanoseconds>(amount);
       } else if (unit == "us") {
-        result = toDuration.template operator()<microseconds>(amount);
+        return toDuration.template operator()<microseconds>(amount);
       } else if (unit == "ms") {
-        result = toDuration.template operator()<milliseconds>(amount);
+        return toDuration.template operator()<milliseconds>(amount);
       } else if (unit == "s") {
-        result = toDuration.template operator()<seconds>(amount);
+        return toDuration.template operator()<seconds>(amount);
       } else if (unit == "min") {
-        result = toDuration.template operator()<minutes>(amount);
+        return toDuration.template operator()<minutes>(amount);
       } else if (unit == "h") {
-        result = toDuration.template operator()<hours>(amount);
+        return toDuration.template operator()<hours>(amount);
       } else {
         // Unsupported suffix
         AD_FAIL();
       }
-    } else {
-      is.setstate(is.rdstate() | std::ios_base::failbit);
     }
-
-    return is;
+    throw std::runtime_error{
+        absl::StrCat("Failed to convert string '", arg, "' to duration type.")};
```

Comment:
Can this error message include some examples for valid strings. ("Examples for valid duration strings are `4s`, `30129ms` etc.")